contentful/apps
# feat: display MS org name logo [INTEG-1782]

## Purpose

This PR updates the access section of the MS Teams app config page to display the organization name and logo after a user authenticates with Microsoft. 

<img width="928" alt="Screenshot 2024-02-14 at 9 38 06 AM" src="https://github.com/contentful/apps/assets/62958907/7eed5f7a-ab8b-4cce-a91d-1c25a6577c51">

## Approach

While the tenant Id and the username can be accessed from the MSAL hook, we need to call the MS Graph API in order to get the organization name and logo. I created a new `msGraphApi` util class to work with the API. This class handles getting the access token for the API request, calling the correct endpoints, and error handling. 

If there is an error getting the organization name and logo, these items won't display and a notification will show saying that there is error getting organization details

![Screenshot 2024-02-14 at 8 45 38 AM](https://github.com/contentful/apps/assets/62958907/5f4a67a5-da47-4910-a44b-0921e183cf9d)

In order to create tests for `msGraphApi`, I utilized MSW to mock the HTTP requests. 

## Testing steps

Navigate to the MS Teams development space and authenticate to our sandbox account to see the org name and logo.

## Breaking Changes

## Dependencies and/or References

See my notes in confluence for more information about working with the MS Graph API.

## Deployment
